system/src/Grav/Common/User/User.php

```diff
@@ -192,9 +192,9 @@ public function save()
         if ($file) {
             $username = $this->get('username');
 
-            if (!$file->filename()) {
+            if (!$file->filename() || $username != mb_strtolower($username)) {
```

Comment:
Is this extra check for backwards compat ?

Reply:
If my memory doesn't fail me, this was meant to guarantee the creation of lowercase-named files, which could be named uppercase in the case of the files having been already assigned at this point, causing a handful of problems.